### ISSUE TITLE
Getting started

### ISSUE THREAD

**Issue by jimcbrodie** [RobinShipston/Tesla-Powerwall#1]
Hi Robin,
Arrived here from your reply to my question on the Tesla forum. 
I see I'm embarking on a voyage of discovery. 
It's a start pretty much from scratch, so a nudge in the right direction to begin with would be much appreciated! How do I run these groovy scripts (on Android, ideally)? 
Cheers, 
Jim

**Comment by RobinShipston** [comment]
These scripts are only for creating a device driver and app for the SmartThings hub. You’ll need one of those to use these. Although the code will show you how to query the Powerwall if you want to create something in a different environment, but if you are starting from scratch then you’ve got a bit of a learning curve. A SmartThings hub is perhaps the easiest route right now, but of course Amazon and Google have entered the ring, so that may change.

**Comment by jimcbrodie** [comment]
Thanks Robin.

I didn't realise this was specific to the SmartThings hub. I'm hoping it
might be possible to set this up with IFTTT. Ideally without having to buy
any more equipment!
I have a couple of 'smart plugs' on order, which will make the heaters
networkable. At worst, it'll mean I can switch them on/off from the tablet,
without having to go to the sockets every time.

**Comment by RobinShipston** [comment]
You will have to open your Powerwall up to the internet unsecured to get IFTTT to work. The Powerwall can be turned on or off without authentication, and thanks to Tesla not sharing any information, no one knows quite how secure it is and if it could give access to your home network. Just a heads-up!

**Comment by jimcbrodie** [comment]
Hmm. Okay. It would be nice to have an intermediary to query the Powerwall,
without making it publicly available. That's where I was hoping your app
would come in!

**Comment by RobinShipston** [comment]
Yeah, if Tesla would just open up the api that their app uses, things would be much easier. They just have zero interest in playing nicely with anyone else.

**Comment by jimcbrodie** [comment]
I bought a Lenovo Tab not long ago, which came with an Alexa app I have
ignored completely so far (uninstalled, in fact). Perhaps I'll take a look
and see if it might be any help here.

**Comment by jimcbrodie** [comment]
Hi Robin,
Unsurprisingly, the Alexa app was a non-starter. 
I have since come across this, which looks very promising: https://www.home-assistant.io/ 
Seems there's no way around the need for more hardware, so I've ordered a Raspberry Pi to run it on. 
Your code for querying the Powerwall should be ideal! 
Cheers, 
Jim
